File: 2015/day02.lua
require "utils"
local input = utils.split(utils.readFile("day02input.txt"), "[^\n]+")

local function answer1(input)
	local result = 0
	for _, line in ipairs(input) do
		local dim = utils.split(line, "[^x]+") -- L, W, H
		local minArea
		for i, v in ipairs(dim) do
			for i2, v2 in ipairs(dim) do
				if not (i == i2) then
					if not minArea or minArea > v*v2 then
						minArea = v*v2
					end
				end
			end
		end
		result = result + (2 * dim[1] * dim[2]) + (2 * dim[3] * dim[2]) + (2 * dim[3] * dim[1]) + minArea
	end
	return result
end

local function answer2(input)
	local result = 0
	for _, line in ipairs(input) do
		local dim = utils.split(line, "[^x]+") -- L, W, H
		dim[1], dim[2], dim[3] = tonumber(dim[1]), tonumber(dim[2]), tonumber(dim[3])
		table.sort(dim, function(a,b) return a < b end)
		result = result + (dim[1]*2) + (dim[2]*2) + (dim[1] * dim[2] * dim[3])
	end
	return result
end

print("the answer 1 is", answer1(input))
print("the answer 2 is", answer2(input))

File: 2015/ruby/day02.rb
$input = File.open("day02.txt", "r").readlines

def answer1()
	result = 0
	$input.each do |line|
		present = line.split("x").map(&:to_i)
		result += 2 * present[0] * present[1]
		result += 2 * present[1] * present[2]
		result += 2 * present[2] * present[0]
		present.sort!
		result += present[0] * present[1]
	end
	result
end

def answer2()
	result = 0
	$input.each do |line|
		present = line.split("x").map(&:to_i)
		present.sort!
		result += (present[0] + present[1]) * 2 + present.inject(:*)
	end
	result
end

puts "the answer 1 is " + answer1().to_s
puts "the answer 2 is " + answer2().to_s